src/main/java/org/kuali/kfs/sys/document/service/impl/AccountingLineRuleHelperServiceImpl.java

```diff
@@ -0,0 +1,589 @@
+    }
+
+    @Override
+    public boolean isValidSubAccount(String accountIdentifyingPropertyName, SubAccount subAccount) {
+        return isValidSubAccount(subAccount, KFSPropertyConstants.SUB_ACCOUNT_NUMBER, accountIdentifyingPropertyName);
+    }
+
+    @Override
+    public boolean isValidSubAccount(SubAccount subAccount, String errorPropertyName, String errorPropertyIdentifyingName) {
+        List<AccountingLineValidationError> errors = getSubAccountValidationErrors(subAccount, errorPropertyName, errorPropertyIdentifyingName, false);
+        setGlobalErrorMessages(errors);
+        return errors.isEmpty();
+    }
+
+    protected List<AccountingLineValidationError> getSubAccountValidationErrors(SubAccount subAccount, String errorPropertyName,
+                                                                             String errorPropertyIdentifyingName, boolean useShortMessages) {
+        String label = getSubAccountLabel();
+
+        // Check exists & active
+        // CORNELL FIX: boolean isActive = subAccount != null && subAccount.isActive();
+        boolean isActive = ObjectUtils.isNotNull(subAccount) && subAccount.isActive();
+        return getExistsActiveValidationErrors(subAccount, isActive, useShortMessages, errorPropertyName, errorPropertyIdentifyingName, label);
+    }
+
+    @Override
+    public boolean isValidSubObjectCode(String accountIdentifyingPropertyName, SubObjectCode subObjectCode) {
+        return isValidSubObjectCode(subObjectCode, KFSPropertyConstants.FINANCIAL_SUB_OBJECT_CODE, accountIdentifyingPropertyName);
+    }
+
+    @Override
+    public boolean isValidSubObjectCode(SubObjectCode subObjectCode, String errorPropertyName, String errorPropertyIdentifyingName) {
+        List<AccountingLineValidationError> errors = getSubObjectCodeValidationErrors(subObjectCode, errorPropertyName, errorPropertyIdentifyingName, false);
+        setGlobalErrorMessages(errors);
+        return errors.isEmpty();
+    }
+
+    protected List<AccountingLineValidationError> getSubObjectCodeValidationErrors(SubObjectCode subObjectCode, String errorPropertyName,
+            String errorPropertyIdentifyingName, boolean useShortMessages) {
+        String label = getSubObjectCodeLabel();
+
+        // Check exists & active
+        // CORNELL FIX: boolean isActive = subObjectCode != null && subObjectCode.isActive();
+        boolean isActive = ObjectUtils.isNotNull(subObjectCode) && subObjectCode.isActive();
+        return getExistsActiveValidationErrors(subObjectCode, isActive, useShortMessages, errorPropertyName, errorPropertyIdentifyingName, label);
+    }
+
+    /**
+     * Validates an accounting line. Side effects: refreshes referenced BOs and sets global error messages
+     *
+     * @param accountingLine
+     * @return boolean whether the accounting line is valid
+     */
+    @Override
+    public boolean validateAccountingLine(AccountingLine accountingLine) {
+        if (accountingLine == null) {
```

Comment:
Another spot that may need the ObjectUtils update, despite not being changed by FINP-7183.

Reply:
Added null check.